src/libs/tools/src/errors/errorFactory.cpp

```diff
@@ -0,0 +1,150 @@
+
+#include <keyset.hpp>
+#include <kdberrors.h> // for code and description constants
+#include "errors/errorFactory.hpp"
+#include "errors/warningFactory.hpp"
+#include "errors/errorTypes.hpp"
+
+namespace kdb
+{
+namespace tools
+{
+namespace errors
+{
+
+Error* ErrorFactory::create(const std::string & type, const std::string & reason, const std::string & module,
+				  const std::string & file, const std::string & mountPoint, const std::string & configFile, kdb::long_t line)
+{
+	if (type == ELEKTRA_ERROR_RESOURCE || type == ELEKTRA_ERROR_RESOURCE_NAME)
```

Comment:
Why would be the type sometimes the _NAME and sometimes not?

Reply:
At the bottom of [`/src/include/kdberrors.h`](https://github.com/ElektraInitiative/libelektra/blob/master/src/include/kdberrors.h), for every introduced code four constants are defined:

```[c]
#define DECLARE_ERROR_CODE(cname)                                                                                                          \
	extern const char * ELEKTRA_ERROR_##cname;                                                                                         \
	extern const char * ELEKTRA_ERROR_##cname##_NAME;                                                                                  \
                                                                                                                                           \
	extern const char * ELEKTRA_WARNING_##cname;                                                                                       \
	extern const char * ELEKTRA_WARNING_##cname##_NAME;                                                                                \
                                                                                                                                           \
	void elektraSetError##cname (Key * key, const char * file, const char * line, const char * module, const char * reason, ...);      \
	void elektraAddWarning##cname (Key * key, const char * file, const char * line, const char * module, const char * reason, ...);

DECLARE_ERROR_CODE (RESOURCE)
DECLARE_ERROR_CODE (OUT_OF_MEMORY)
DECLARE_ERROR_CODE (INSTALLATION)
DECLARE_ERROR_CODE (INTERNAL)
DECLARE_ERROR_CODE (INTERFACE)
DECLARE_ERROR_CODE (PLUGIN_MISBEHAVIOR)
DECLARE_ERROR_CODE (CONFLICTING_STATE)
DECLARE_ERROR_CODE (VALIDATION_SYNTACTIC)
DECLARE_ERROR_CODE (VALIDATION_SEMANTIC)

#undef DECLARE_ERROR_CODE
```

We designed to Factories so that the `ELEKTRA_ERROR_##cname` or `ELEKTRA_ERROR_##cname##_NAME` are valid parameters for creating an object of the appropriate type.

Should we change that so only the Code, but not the Description/_NAME is valid here?

Reply:
Yes, I do not think that _NAME should be anything external. It is only needed to render the error text. I created #4230